src/EditorFeatures/CSharpTest2/Recommendations/ScopedKeywordRecommenderTests.cs

```diff
@@ -0,0 +1,424 @@
+@"fixed ($$"));
+        }
+
+        [Fact]
+        public async Task TestNotInDelegateReturnType()
+        {
+            await VerifyAbsenceAsync(
+@"public delegate $$");
+        }
+
+        [Fact]
+        public async Task TestPossibleLambda()
+        {
+            // Could be `var x = ((scoped ref int x) => x);`
+            await VerifyKeywordAsync(AddInsideMethod(
+@"var x = (($$"));
+        }
+
+        [Fact]
+        public async Task TestEmptyStatement()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"$$"));
+        }
+
+        [Fact]
+        public async Task TestBeforeStatement()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"$$
+return true;"));
+        }
+
+        [Fact]
+        public async Task TestAfterStatement()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"return true;
+$$"));
+        }
+
+        [Fact]
+        public async Task TestNotInClass()
+        {
+            await VerifyAbsenceAsync(@"class C
+{
+  $$
+}");
+        }
+
+        [Fact]
+        public async Task TestInFor()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"for ($$"));
+        }
+
+        [Fact]
+        public async Task TestInFor2()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"for ($$;"));
+        }
+
+        [Fact]
+        public async Task TestInFor3()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"for ($$;;"));
+        }
+
+        [Fact]
+        public async Task TestNotInFor()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"for (var $$"));
+        }
+
+        [Fact]
+        public async Task TestNotAfterVar()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"var $$"));
+        }
+
+        [Fact]
+        public async Task TestInForEach()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"foreach ($$"));
+        }
+
+        [Fact]
+        public async Task TestNotInForEach()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"foreach (var $$"));
+        }
+
+        [Fact]
+        public async Task TestInAwaitForEach()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"await foreach ($$"));
+        }
+
+        [Fact]
+        public async Task TestNotInAwaitForEach()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"await foreach (var $$"));
+        }
+
+        [Fact]
+        public async Task TestNotInForEachRefLoop0()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"foreach (ref $$"));
+        }
+
+        [Fact]
+        public async Task TestNotInForEachRefLoop1()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"foreach (ref $$ x"));
+        }
+
+        [Fact]
+        public async Task TestNotInForEachRefLoop2()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"foreach (ref v$$ x"));
```

Comment:
>ref v$$ [](http://example.com/codeflow?start=11&length=7)

Should this be `ref s$$`?

Reply:
Nice catch! @cston 
I originally started with `var` tests.